bin/embark

```diff
@@ -1,15 +1,1023 @@
 #!/usr/bin/env node
 
-try {
-  eval('let __nodeTest = 123;');
-} catch(e) {
-  if (e.name === 'SyntaxError') {
-    console.error("unsupported version of NodeJS. Make sure you are running nodejs 8.11.3 or above");
+/* global __dirname __filename process require */
 
-    process.exit();
+// This script doesn't use JS syntax, packages, or APIs *un*supported by any
+// node version >=4.0.0, so unsupported versions from v4.0.0+ will get the
+// intended error messages. A future version of this script could instead rely
+// on babel to achieve the same goal.
+// See: https://node.green/
+
+// KEY ASSUMPTION: for a DApp to be valid, from embark's perspective, it must
+// have a parsable embark.json file in its top-level directory; if that
+// requirement changes in the future then this script must be revised.
+// Hypothetical example of such a change: embark config info may be included in
+// package.json under `{"embark": {...}}` -or- stored in an embark.json file.
+
+function main() {
+  whenNoShim();
+  var invoked = thisEmbark();
+  var embarkJson = findEmbarkJson();
+  var dappPath = embarkJson.dirname;
+  process.chdir(dappPath);
+  process.env.DAPP_PATH = dappPath;
+  process.env.PWD = dappPath;
```

Comment:
Do we still need to set again those 2 env var in the controller? (same question below for other env var)
I think now they are all defined twice

Reply:
I left them in the controller thinking someone might import that module manually, to execute embark commands programmatically, in which case it will be important that the env vars are set with some defaults. If that scenario is too unlikely, I can remove them from the controller. Thoughts?

Reply:
I personally don't do that, it might be something we ask the team.